gspread/spreadsheet.py

```diff
@@ -739,3 +746,32 @@ def list_protected_ranges(self, sheetid):
             raise WorksheetNotFound("worksheet id {} not found".format(sheetid))
 
         return sheet.get("protectedRanges", [])
+
+    def get_all_worksheet_values(self, skip_worksheet_titles: list[str] = None):
+        """Grabs all the data from all the worksheets in one API call. Skips any worksheets that were named in the
+        skip_worksheet_title param.
+        :returns Dict of worksheet data with worksheet title as key
+        """
+
+        if skip_worksheet_titles is None:
+            skip_worksheet_titles = []
+
+        ranges = []
+
+        for worksheet in self.google_sheet.worksheets():
+            if worksheet.title not in skip_worksheet_titles:
+                ranges.append(
+                    f"{worksheet.title}!A1:{column_index_to_letter(worksheet.col_count)}"
+                )
```

Comment:
We have utils function to compute a full range, here you don't need to give it a range. you can give an empty range the API will return the whole sheet with all data.